test/IISIntegration.FunctionalTests/Utilities/SkipIfNotAllowedToRunIISAttribute.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Security.Principal;
+using Microsoft.AspNetCore.Testing.xunit;
+using Microsoft.Win32;
+
+namespace Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests
+{
+    [AttributeUsage(AttributeTargets.Assembly | AttributeTargets.Class | AttributeTargets.Method)]
+    public sealed class SkipIfNotAllowedToRunIISAttribute : Attribute, ITestCondition
+    {
+        public bool IsMet {
+            get
+            {
+                // Two conditions need to be met:
+                // IIS needs to be installed
+                var identity = WindowsIdentity.GetCurrent();
+                var principal = new WindowsPrincipal(identity);
+                var isInrole = principal.IsInRole(WindowsBuiltInRole.Administrator);
```

Comment:
There's a difference between being an administrator and running elevated. Does this detect elevation?

Reply:
It doesn't. Not sure if we care though?

Reply:
Won't the test fail without elevation?

Reply:
You cannot enable these tests if they're going to fail when running non-elevated. Full Stop.

Is that why the AppVeyor tests are failing?

Reply:
Resolved offline. WindowsBuiltInRole.Administrator returns true only if your current role is elevated. The skip condition works for skipping builds when running non-admin.